File: 2015/gara3/magneti/att/magneti.c
#include <stdio.h>
#include <assert.h>

#define MAXN 100000

int gira(int N, char descrizione[]) {
    // Mettete qui il codice della soluzione
    return 42;
}


char descrizione[MAXN + 5]; // Maggiore di MAXN per contenere newline e null

int main() {
    FILE *fr, *fw;
    int N;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(1 == fscanf(fr, "%d\n", &N));
    assert(fgets(descrizione, MAXN + 5, fr) == descrizione);
    descrizione[N] = 0;

    fprintf(fw, "%d\n", gira(N, descrizione));
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed1/gara3/magneti/att/magneti.pas
function gira(N: longint; var descrizione: array of char): longint;
begin
    (* Mettete qui il codice della soluzione *)
    gira := 42;
end;


const
    MAXN = 100000;
var
    N           : longint;
    descrizione : array[0..MAXN] of char;
    fr, fw      : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N);
    readln(fr, descrizione);

    writeln(fw, gira(N, descrizione));
    close(fr);
    close(fw);
end.
